posts/2021/12/conda-recipe-grayskull.md

```diff
@@ -0,0 +1,118 @@
+From here onwards in this blog we’ll assume that we want to publish our package on the conda-forge channel.
+
+## Publishing a package on conda-forge
+Publishing packages on the conda-forge channel requires the knowledge of ‘recipes’. A recipe is a collection of files that defines how to build a package. Minimally a recipe contains a meta.yaml file that describes:
+- The package name and version
+- its dependencies
+- how to build it
+- some other metadata
+You can learn more about recipes [here](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#meta-yaml).
+
+<p align="center">
+    <img
+     alt="Cute representation of a package recipe"
+     src="/images/2021/12/recipe_animation.png">
+    <i><br>Cute representation of a package recipe</i>
+</p>
+
+To publish your package write its recipe and create a pull request on the [staged-recipes](https://github.com/conda-forge/staged-recipes) repository of conda-forge. This pull request will be community reviewed and if approved, your package will become available on the conda-forge channel.
+Under the hood, the recipe that you submit is fed to Conda-build which ultimately generates the package.
+
+<p align="center">
+    <img
+     alt="Conda-build transforms recipes into packages"
+     src="/images/2021/12/conda-build.png">
+    <i><br>Conda-build transforms recipes into packages</i>
+</p>
+
+## Grayskull — the automatic Conda recipe generator
+The process to publish a package on conda-forge is simple and straight forward: write the recipe, create a pull request, wait for the community review. But writing recipes is not simple. It can be error prone and tiresome.
+To alleviate this conda-forge provides a template recipe that can be used as a starting point and edited according to one’s needs. But even that could be too intimidating for someone new to packaging and recipes.
+
+Grayskull solves this problem. Grayskull is an automatic conda recipe generator, with a focus on conda-forge. It generates concise and accurate recipes very quickly, provided the package is available on PyPI.
+All you have to do is pass in the name of the Python package to Grayskull and it will generate its recipe for you.
+
+<p align="center">
+    <img
+     alt="Grayskull automates recipe generation"
+     src="/images/2021/12/package_name.png">
+    <i><br>Grayskull automates recipe generation</i>
+</p>
+
+Now that you have the package recipe, you create a pull request on the staged-recipes repository and wait for someone from the conda-forge community to review it. You know the drill!
+
+## But what if a package is not published on PyPI?
+Yes. Unfortunately that’s where Grayskull falls a little short. It only generates recipes for Python packages available on [PyPI](https://pypi.org/). This prerequisite leaves out a number of Python packages otherwise available online.
```

Comment:
```suggestion
Yes. Unfortunately, that’s where Grayskull falls a little short. It only generates recipes for Python packages available on [PyPI](https://pypi.org/). This prerequisite leaves out a number of Python packages otherwise available online.
```